<launch>
    <param name="debug_flag" type="int" value="2" /> <!-- Parameter to specify print level. -->
    <param name="ros_loop_rate" type="int" value="20" /> <!-- Parameter to specify loop rate. -->
    <param name="filter_window" type="int" value="1" /> <!-- Parameter to specify mean-filter window for acceleration. -->
    <param name="mass" type="double" value="1.2" /> <!-- Parameter to specify mass of the drone. -->
    
    <param name="regression_on_sub_topic" type="string" value="/regression_on" /> <!-- Parameter to specfy topic name for regression switch. -->
    <param name="nmpc_cmd_attitude_sub_topic" type="string" value="/outer_nmpc_cmd/rpy" /> <!-- Parameter to specfy topic name for attitude commands from high-level controller. -->
    <param name="nmpc_cmd_thrust_sub_topic" type="string" value="/outer_nmpc_cmd/Fz_FzScaled" /> <!-- Parameter to specfy topic name for thrust commands from high-level controller. -->
    <param name="vel_sub_topic" type="string" value="/mavros/mocap/velocity_body" /> <!-- Parameter to specfy topic name for velocity feedback. -->
    <param name="imu_sub_topic" type="string" value="/mavros/imu/data" /> <!-- Parameter to specfy topic name for IMU feedback. -->


    <param name="gp_model_type" type="int" value="0" /> <!-- Parameter to set gp_type. 0 for concatenated-type.-->
    <param name="max_initial_rec_time_sec" type="int" value="10" /> <!-- Paramter to set maximum initial recording time.-->
    <param name="num_window_points_past" type="int" value="20" /> <!-- Paramter to set number of window points in past.-->
    <param name="num_predict_points_future" type="int" value="1" /> <!-- Paramter to set number of prediction points in future.-->
    <param name="rec_rate_factor" type="double" value="0.6" /> <!-- Paramter to set data recording rate factor. Data recoring rate is set equal to rec_rate_factor/sampleTime. -->

    <param name="gp_input_data_type" type="int" value="0" /> <!-- Parameter to set gp input data type. 0 for type0, 1 for type1.-->
    <param name="feature_data_model_type" type="int" value="0" /> <!-- Parameter to set feature data model type. 0 for subscribed, 1 for compute.-->
    <param name="use_model_difference" type="bool" value="false" /> <!-- Paramter to specify whether to use model difference values in feature data.-->
    <param name="use_states" type="bool" value="false" /> <!-- Paramter to specify whether to use state values in feature data.-->
    <param name="use_controls" type="bool" value="false" /> <!-- Paramter to specify whether to use control values in feature data.-->
    <param name="num_features" type="int" value="3" /> <!-- Paramter to set number of features. Note that this parameter is only for feature_data_model_type = 0, otherwise, it is calculated.-->
    <param name="data_history" type="int" value="5" /> <!-- Paramter to set input data history.-->

    <param name="sf_t0" type="double" value="1.0" /> <!-- Paramter to set sf_t0. -->
    <param name="sn_t0" type="double" value="1.0" /> <!-- Paramter to set sn_t0. -->
    <param name="max_gradient_updates" type="int" value="200" /> <!-- Paramter to set the maximum number of gradient updates for obtaining loghyper minimum.-->
    <param name="covfun1" type="string" value="CovSEard" /> <!-- Paramter to set first covariance function.-->
    <param name="covfun2" type="string" value="CovNoise" /> <!-- Paramter to set second covariance function.-->
    <param name="hyper_optimize_method" type="string" value="CG" /> <!-- Paramter to set optimization method. CG or RPROP-->

    <param name="mean_t0" type="double" value="0.0" /> <!-- Paramter to set mean_t0. -->
    <param name="variance_t0" type="double" value="100.0" /> <!-- Paramter to set variance_t0. -->

</launch>